File: app/src/main/java/com/example/worldvisit/database/DatabaseHelper.java
package com.example.worldvisit.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

import java.util.ArrayList;

public class DatabaseHelper extends SQLiteOpenHelper {

    public static String DATABASE_NAME = "pays_database";
    private static final int DATABASE_VERSION = 1;
    private static final String TABLE_pays = "pays";
    private static final String KEY_ID = "id";
    private static final String NAME = "name";
    private static final String CAPITALE = "capital";
    private static final String CONTINENT = "name";
    private static final String DATE = "date";

    /*CREATE TABLE pays ( id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, phone_number TEXT......);*/

    private static final String CREATE_TABLE_pays = "CREATE TABLE "
            + TABLE_pays + "(" + KEY_ID
            + " INTEGER PRIMARY KEY AUTOINCREMENT," + NAME + CAPITALE + CONTINENT + DATE + " TEXT );";

    public DatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);

        Log.d("table", CREATE_TABLE_pays);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_TABLE_pays);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS '" + TABLE_pays + "'");
        onCreate(db);
    }

    public long addpaysDetail(String name, String capital, String region, String date) {
        SQLiteDatabase db = this.getWritableDatabase();
        // Creating content values
        ContentValues values = new ContentValues();
        values.put(NAME, name);
        values.put(CAPITALE, capital);
        values.put(CONTINENT, region);
        values.put(DATE, date);
        // insert row in pays table
        long insert = db.insert(TABLE_pays, null, values);

        return insert;
    }

    public ArrayList<String> getAllpaysList() {
        ArrayList<String> paysArrayList = new ArrayList<String>();
        String name="";
        String capital="";
        String region="";
        String date="";
        String selectQuery = "SELECT  * FROM " + TABLE_pays;
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor c = db.rawQuery(selectQuery, null);
        // looping through all rows and adding to list
        if (c.moveToFirst()) {
            do {
                name = c.getString(c.getColumnIndex(NAME));
                capital = c.getString(c.getColumnIndex(CAPITALE));
                region = c.getString(c.getColumnIndex(CONTINENT));
                date = c.getString(c.getColumnIndex(DATE));
                // adding to pays list
                paysArrayList.add(name);
                paysArrayList.add(capital);
                paysArrayList.add(region);
                paysArrayList.add(date);
            } while (c.moveToNext());
            Log.d("array", paysArrayList.toString());
        }
        return paysArrayList;
    }

}

File: app/src/main/java/com/example/worldvisit/database/DatabaseHelper.kt
package com.example.worldvisit.database

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log
import com.example.worldvisit.ws.Pays
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.*

class DatabaseHelper(context: Context?) : SQLiteOpenHelper(
    context,
    DATABASE_NAME,
    null,
    DATABASE_VERSION
) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_TABLE_pays)
    }

    override fun onUpgrade(
        db: SQLiteDatabase,
        oldVersion: Int,
        newVersion: Int
    ) {
        db.execSQL("DROP TABLE IF EXISTS '$TABLE_pays'")
        onCreate(db)
    }

    fun addpaysDetail(
        name: String?,
        capital: String?,
        region: String?,
        date: String?,
        short_code: String?
    ): Long {
        val db = this.writableDatabase
        // Creating content values
        val values = ContentValues()
        values.put(NAME, name)
        values.put(CAPITALE, capital)
        values.put(CONTINENT, region)
        values.put(DATE, date)
        values.put(SHORT_CODE, short_code)
        // insert row in pays table
        return db.insert(
            TABLE_pays,
            null,
            values
        )
    }// adding to pays list


    // looping through all rows and adding to list
    val allpaysList: ArrayList<Pays>
        get() {
           val selectQuery =
                "SELECT  * FROM $TABLE_pays"
            val db = this.readableDatabase
            val c = db.rawQuery(selectQuery, null)
            // looping through all rows and adding to list
            val pays = ArrayList<Pays>()

            if (c.moveToFirst()) {
                do {
                    pays.add(
                        Pays(
                            c.getString(c.getColumnIndex(NAME)),
                            c.getString(c.getColumnIndex(CAPITALE)),
                            c.getString(c.getColumnIndex(CONTINENT)),
                            c.getString(c.getColumnIndex(DATE)),
                            c.getString(c.getColumnIndex(SHORT_CODE)),
                            "http://www.geognos.com/api/en/countries/flag/" + c.getString(c.getColumnIndex(SHORT_CODE)).toString().toUpperCase() + ".png"
                        )
                    );
                } while (c.moveToNext())
                //Log.d("array", pays.toString())
            }
            return pays
        }

    companion object {
        var DATABASE_NAME = "pays_database"
        private const val DATABASE_VERSION = 1
        private const val TABLE_pays = "pays"
        private const val KEY_ID = "id"
        private const val NAME = "name"
        private const val CAPITALE = "capital"
        private const val CONTINENT = "region"
        private const val DATE = "date"
        private const val SHORT_CODE = "short_code"

        /*CREATE TABLE pays ( id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, phone_number TEXT......);*/
        private const val CREATE_TABLE_pays = ("CREATE TABLE "
                + TABLE_pays + "("
                + KEY_ID + " INTEGER PRIMARY KEY AUTOINCREMENT,"
                + NAME + " varchard(50),"
                + CAPITALE + " varchard(50),"
                + CONTINENT + " varchard(50),"
                + DATE + " TEXT,"
                + SHORT_CODE + " varchard(50)"
                + ");")
    }

    init {
        Log.d(
            "table",
            CREATE_TABLE_pays
        )
    }
}